scripts/tests/test_zkai_native_attention_mlp_adapter_compression_ablation_gate.py

```
@@ -0,0 +1,154 @@
+            gate.validate_current_sources(mutated)
+
+        mutated = copy.deepcopy(sources)
+        mutated["gate"]["routes"]["adapter_boundary"]["adapter_trace_cells"] += 1
+        with self.assertRaisesRegex(gate.AdapterCompressionAblationError, "adapter route adapter_trace_cells mismatch"):
+            gate.validate_current_sources(mutated)
+
+    def test_current_source_validation_pins_raw_artifact_hashes(self) -> None:
+        sources = gate.load_sources()
+        mutated = copy.deepcopy(sources)
+        mutated["source_artifacts"][0]["sha256"] = "00" * 32
+        with self.assertRaisesRegex(gate.AdapterCompressionAblationError, "current_single_proof_gate hash drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)